modules/@apostrophecms/ui/ui/apos/utils/index.js

```diff
@@ -40,25 +76,46 @@ function debounceAsync(fn, delay, options = {}) {
 
       previousDone = true;
       res(returned);
-    }, delay);
+    }
+    if (delay === false) {
+      return body();
+    } else {
+      return setTimeout(body, delay);
+    }
   };
 
-  function wrapper(...args) {
+  let wrapper = (...args) => {
     return new Promise((resolve, reject) => {
       if (canceled) {
         return reject(canceledRejection);
       }
+      if (skipNextDelay) {
+        skipNextDelay = false;
+        setTimer(resolve, reject, args, false);
+      }
       clearTimeout(timer);
       timer = setTimer(resolve, reject, args, delay);
     });
   };
 
+  if (options.ifNotCanceled) {
+    wrapper = wrapper().then(...args => {
+      options.ifNotCanceled(...args);
```

Comment:
Must be synchronous, has no job except side effects, so we intentionally pay no attention to its return value.